### ISSUE TITLE
VS code very slow executing PowerShell scripts

### ISSUE THREAD

**Issue by ProfMars** [microsoft/vscode#107971]
Issue Type: <b>Performance Issue</b>

I have the issue that VSCode gets very slow. This is especially noticable in the Terminal pane where the output is displayed. Whilst running a script in normale powershell window or Windows Terminal it flies across the screen. In VSCode it does not.
I disabled alle extensions (except the PowerShell extension), but the same issue occours.
This renders VSCode useless for me at this time and I need to move back to ISE to my regrief.
Please help me get this fixed.

VS Code version: Code 1.49.2 (e5e9e69aed6e1984f7499b7af85b3d05f9a6883a, 2020-09-24T16:29:41.983Z)
OS version: Windows_NT x64 10.0.20226

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Xeon(R) E-2286M  CPU @ 2.40GHz (16 x 2400)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>opengl: enabled_on<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|31.85GB (16.60GB free)|
|Process Argv|--crash-reporter-id d9de75d6-89e7-4d50-a08c-488fc8b5921d|
|Screen Reader|yes|
|VM|0%|
</details><details>
<summary>Process Info</summary>

```
CPU %	Mem MB	   PID	Process
    1	   130	 16268	code main
    0	    98	  5752	   shared-process
    6	   315	 15536	   window (ServicedeskPortal-AzureServiceHealthIncidentInterface.ps1 - PowerShell Scripts - Visual Studio Code)
    0	   230	 10780	     C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive -ExecutionPolicy Bypass -Command "Import-Module 'c:\Users\mheek\.vscode\extensions\ms-vscode.powershell-2020.6.0\modules\PowerShellEditorServices\PowerShellEditorServices.psd1'; Start-EditorServices -HostName 'Visual Studio Code Host' -HostProfileId 'Microsoft.VSCode' -HostVersion '2020.6.0' -AdditionalModules @('PowerShellEditorServices.VSCode') -BundledModulesPath 'c:\Users\mheek\.vscode\extensions\ms-vscode.powershell-2020.6.0\modules' -EnableConsoleRepl -StartupBanner \"=====> PowerShell Integrated Console v2020.6.0 <=====
\" -LogLevel 'Normal' -LogPath 'c:\Users\mheek\.vscode\extensions\ms-vscode.powershell-2020.6.0\logs\1601643239-a9720243-4d34-4ec4-b0b6-0d3dd29cba7b1601643190332\EditorServices.log' -SessionDetailsPath 'c:\Users\mheek\.vscode\extensions\ms-vscode.powershell-2020.6.0\sessions\PSES-VSCode-16268-273305' -FeatureFlags @() "
    0	     7	 18136	     console-window-host (Windows internal process)
    0	    99	 24376	     extensionHost
    0	    69	 35720	       "C:\Users\mheek\AppData\Local\Programs\Microsoft VS Code\Code.exe" "c:\Users\mheek\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\json-language-features\server\dist\node\jsonServerMain" --node-ipc --clientProcessId=24376
    0	     7	 29712	     console-window-host (Windows internal process)
    0	   150	 31456	     C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe
    0	    16	 35140	     watcherService 
    0	    11	 32628	       console-window-host (Windows internal process)
    0	   100	 23232	   gpu-process
    0	    85	 23268	   window (Issue Reporter)
    0	    25	 24104	   crashpad-handler
    0	    41	 33628	   utility
```

</details>
<details>
<summary>Workspace Info</summary>

```
|  Window (ServicedeskPortal-AzureServiceHealthIncidentInterface.ps1 - PowerShell Scripts - Visual Studio Code)
|    Folder (PowerShell Scripts): 311 files
|      File types: ps1(205) json(6) log(5) txt(2) gitignore(1) suo(1)
|                  sqlite(1) vsdx(1) png(1) pdf(1)
|      Conf files: launch.json(1)
|      Launch Configs: PowerShell(5);
```

</details>
<details><summary>Extensions (1)</summary>

Extension|Author (truncated)|Version
---|---|---
powershell|ms-|2020.6.0

</details>

**Comment by meganrogge** [comment]
can you pls provide an example of a script that causes VS code to be very slow? @ProfMars

**Comment by Curtmcgirt** [comment]
forgive me, i'm not a developer.  but how is this powershell-specific issue a duplicate of #74620?

the following command takes less than a second in powershell ise but takes 63 seconds in vscode. will a "node-pty host process with flow control and batching" fix this?

Get-DnsServerResourceRecord -RRType A -ComputerName domain.com -ZoneName domain.com | Where-Object {$_.recorddata.ipv4address -like $ip} | Select-Object -ExpandProperty hostname

**Comment by Tyriar** [comment]
tldr, it's complicated.

Because of the fact that the renderer processes' main thread is busy which slows the node.js event loop which causes it to throw data events at a much slower rate. The reason why this is much more likely to happen on Windows and PowerShell specific is related to how the number of events the pty emulation on Windows fires compared to non-Windows and non-pwsh. Moving this out into a separate process (https://github.com/microsoft/vscode/issues/74620) would free the process to focus on the incoming and outgoing data events to the terminal's shell, this is captured in this point:

> Data events from node-pty seems to be blocked by the event loop, I suspect this is main the reason conpty is slow, it may also be slowing down non-Windows data flow too
